src/components/TopicList.jsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from "react";
+import { Link } from "react-router-dom";
+import { getTopics } from "../utils/api";
+
+function TopicList() {
+  const [topics, setTopics] = useState([]);
+
+  useEffect(() => {
+    getTopics().then(({ topics }) => {
+      setTopics(topics);
+    });
+  }, [topics]);
+
+  return (
+    <div>
+      {topics.map((topic) => {
+        return (
+          <Link to={`/articles/${topic.slug}`} key={topic.slug}>
+            <button className="topic-button">{topic.slug}</button>
```

Comment:
I wouldn't put a button inside of a link - semantically this isn't good and it could lead to some strange behaviours. If you want it styled like a button I would place a className on the Link and style it this way.